.cmake-format.py

```diff
@@ -0,0 +1,212 @@
+# ----------------------------------
+# Options affecting listfile parsing
+# ----------------------------------
+with section("parse"):
+
+  # Specify structure for custom cmake functions
+  additional_commands = { 'foo': { 'flags': ['BAR', 'BAZ'],
```

Comment:
We should specify this, at least for our custom functions that use `cmake_parse_arguments`. Something like:

```

    additional_commands = {
        "add_enclave_lib": {
            "kwargs": {"SRCS": "*", "INCLUDE_DIRS": "*", "LINK_LIBS": "*"},
        },
        "add_client_exe": {
            "kwargs": {"SRCS": "*", "INCLUDE_DIRS": "*", "LINK_LIBS": "*"},
        },
        "add_e2e_test": {
            "kwargs": {
                "NAME": "*",
                "PYTHON_SCRIPT": "*",
                "IS_SUITE": "*",
                "CURL_CLIENT": "*",
                "ADDITIONAL_ARGS": "*",
            },
        },
        "add_perf_test": {
            "kwargs": {
                "NAME": "*",
                "PYTHON_SCRIPT": "*",
                "CLIENT_BIN": "*",
                "VERIFICATION_FILE": "*",
                "LABEL": "*",
                "ADDITIONAL_ARGS": "*",
            },
        },
        "add_e2e_test": {
            "kwargs": {"SRCS": "*", "INCLUDE_DIRS": "*", "LINK_LIBS": "*",},
        },
    }
```

(though `add_enclave_lib` will be renamed by #715)

Reply:
Nice!